src/challenges.js

```diff
@@ -1,21 +1,52 @@
+/* eslint-disable*/
 /* ======================== CallBacks Practice ============================ */
 const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
+	  for (let i = 0; i < elements.length; i++) { // //this each function takes a for loop and interates
+				cb(elements[i], i);           // though the elements of the elements array and passes its values to cb one by one.
+                                     //  cb needs to be invoked multiple times once for each value in the array
+                                    //   therefore elements needed an iterator. 
+	}
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
-};
+  const newarray = [];                            
+	 for (let i = 0; i < elements.length; i++) {  // this map function takes a for a loop and interates over the 
+			newarray.push(cb(elements[i]));          //  each values in the array and passes each of the values in 
+			                                         // (elements), map passes each of the values in arr that the for  
+			                                        //   loop iterates and passes all of the values to cb to be invoked map 
+	}                                          //    keeps track of the results invoked by cb and creates a new array 
+	                                          //    (const newarray) containing the values returned of the invocation
+	  return newarray;                       //     of cb that is the same length as the original array and assigns them
+};                                         //     const newarray where they are returned. 
 
 /* ======================== Closure Practice ============================ */
 const limitFunctionCallCount = (cb, n) => {
   // Should return a function that invokes `cb`.
   // The returned function should only allow `cb` to be invoked `n` times.
+    let count = 0;                  // variable count to keep track of how many times cb was invoked. 
+    const limitCall = () => {      //  const limitCall invokes cb and uses the counter to keep track   
+		  cb ();                      //   of how many times cb has been called since the instructions say 
+			count++;                   //   that cb can only be invoked n times. 
+  }
+};
+   return limitCall;           // the variable limitCall returns the results of the invoking cb 'n' times    
 };
 
 const cacheFunction = cb => {
```

Comment:
to access `hasOwnProperty` you would want to do something like `Object.prototype.hasOwnProperty`.  For that line you would actually want to do `Object.prototype.hasOwnProperty.call(Cache, input)`.
Also on line 45/46 you don't need the else statement you can just `return` the item (not really a big deal but the linter was probably warning you about that one.